### PR TITLE
Enable simple NetworkPolicy

### PR DESCRIPTION
**Pull request by smalleni**
Signed-off-by: Sai Sindhur Malleni <smalleni@redhat.com>

### CODE REVIEW

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
Nice one @smalleni!, can we add a test case for this?

File: roles/common/templates/networkpolicy.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,14 @@
+kind: NetworkPolicy
```
Comment (jtaleric) [inline roles/common/templates/networkpolicy.yml.j2:1]: should we name this uperf-networkpolicy, or better yet, should we template the workload name?

Comment (smalleni) [reply roles/common/templates/networkpolicy.yml.j2:1]: That is the kind which is `NetworkPolicy`..the name as you can see below is 

"{{ meta.name }}-networkpolicy-{{ trunc_uuid }}"

Am I missing something?

File: roles/common/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# common tasks across multiple roles go here
+- name: Create Network policy if enabled
+  k8s:
+   definition: "{{ lookup('template', 'networkpolicy.yml.j2') | from_yaml }}"
+  vars:
+    trunc_uuid: "{{ cr_state.resources[0].status.suuid }}"
+  when: workload.args.networkpolicy is defined and workload.args.networkpolicy
```
Comment (dry923) [inline roles/common/tasks/main.yml:7]: As it is written the template would be called every time the reconciliation loop triggered. While it's not really going to break anything we should probably put in a check to see if it exists and then execute if it does not.

something like:
```
block:
  k8s_facts:
      kind: NetworkPolicy
      api_version: networking.k8s.io/v1
      namespace: '{{ operator_namespace }}'
      name: "{{ meta.name }}-networkpolicy-{{ trunc_uuid }}"
    register: net_policy
    
  - name: Create Network policy if enabled
    k8s:
     definition: "{{ lookup('template', 'networkpolicy.yml.j2') | from_yaml }}"
    vars:
      trunc_uuid: "{{ cr_state.resources[0].status.suuid }}"
  when: net_policy | json_query(some_valid_query

 when: workload.args.networkpolicy is defined and workload.args.networkpolicy
```

Thoughts?

File: roles/common/templates/networkpolicy.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: "{{ meta.name }}-networkpolicy-{{ trunc_uuid }}"
+  namespace: '{{ operator_namespace }}'
+spec:
+  podSelector: 
+    matchLabels:
+      type: uperf-bench-server-{{ trunc_uuid }}
```
Comment (jtaleric) [inline roles/common/templates/networkpolicy.yml.j2:9]: The reason I mentioned my comment is because of the "uperf"- here.

File: playbook.yml [hunk 4]
```diff
@@ -56,6 +56,9 @@
     
     when: workload is defined and (cr_state.resources[0].status is not defined or cr_state.resources[0].status.uuid is not defined)
 
+  - include_role:
+      name: "common"
```
Comment (dry923) [inline playbook.yml:60]: Should this be in the lower Run workload block?

File: roles/common/tasks/main.yml (hunk 2 above)
Comment (smalleni) [reply roles/common/tasks/main.yml:7]: Yeh it didn't seem like there was a state that i could take advantage of like Running/Created. I''l check again.

File: roles/common/templates/networkpolicy.yml.j2 (hunk 3 above)
Comment (smalleni) [reply roles/common/templates/networkpolicy.yml.j2]: done

File: roles/common/tasks/main.yml (hunk 2 above)
Comment (smalleni) [reply roles/common/tasks/main.yml:16]: Done

**Review by rsevilla87 — APPROVED** [review]
LGTM!

**Review by dry923 — APPROVED** [review]
LGTM!

**Review by jtaleric — APPROVED** [review]
LGTM

**Pull request merged**